controlplane/kubeadm/internal/etcd/fake/client.go

```diff
@@ -30,13 +30,17 @@ type FakeEtcdClient struct {
 	MemberUpdateResponse *clientv3.MemberUpdateResponse
 	MoveLeaderResponse   *clientv3.MoveLeaderResponse
 	StatusResponse       *clientv3.StatusResponse
+	ErrorResponse        error
 }
 
 func (c *FakeEtcdClient) Endpoints() []string {
 	return c.EtcdEndpoints
 }
 
 func (c *FakeEtcdClient) MoveLeader(_ context.Context, _ uint64) (*clientv3.MoveLeaderResponse, error) {
+	if c.ErrorResponse != nil {
+		return c.MoveLeaderResponse, c.ErrorResponse
+	}
 	return c.MoveLeaderResponse, nil
```

Comment:
```suggestion
	return c.MoveLeaderResponse, c.ErrorResponse
```

should be fine given that `c.ErrorResponse` is nil, same comment for others